### ISSUE TITLE
We need to add support for defining curricula in the lessons app

### ISSUE THREAD

**Issue by timlinux** [kartoza/projecta#740]
See updated schema below which includes the curricula model.

![lessonsapp](https://user-images.githubusercontent.com/178003/35772817-5751f922-094d-11e8-81e1-5b95daff2816.png)

@charlotte-morgan please comment if there are specific attributes you would like to see in the curruciculum form / output report.

**Comment by Gustry** [comment]
Between worksheet and curriculum:
* Each curriculum has zero or more worksheets but also each worksheet can be in zero or more curriculum. So it's becoming a many to many relationship (missing on the drawing)
* Each worksheet in a curriculum can have a different order in the curriculum, so we need another sequence number in this many to many relationship.

**Comment by timlinux** [comment]
Thanks @Gustry - I have updated the diagram accordingly here:

![lessonsapp](https://user-images.githubusercontent.com/178003/35782465-36cf2620-0a01-11e8-993d-ca17e85db588.png)

**Comment by ismailsunni** [comment]
- [ ] We need to implement the front end. :point_up: [February 25, 2018 5:45 AM](https://gitter.im/kartoza/projecta?at=5a91eaefe4ff28713ac11c34)

**Comment by Gustry** [comment]
You need to implement everything, only models are there ;-)

**Comment by ismailsunni** [comment]
Hehe, sure @Gustry 

@timlinux  Is there any discussion about how will the front end look like? If not, perhaps I will create some mock-up before coding

**Comment by timlinux** [comment]
@ismailsunni yes please go ahead and make a mockup and then we can discuss it (and also show it to @charlotte-morgan).

**Comment by haubourg** [comment]
@timlinux Hi Tim, 
if you plan to store personal data of students, please take attention to the European General Data Protection Regulation at https://www.eugdpr.org/ that instores privacy by design.
That may require some user features to allow data download or deletion. Data encryption and explicit user agreement are also mandatory. 
Even if you are hosted outside of EU, as soon as EU citizen are in the database, we are submitted to it.
Cheers Régis

**Comment by timlinux** [comment]
@haubourg thanks for your comment. It does not really relate to the curriculum aspect (which details only how a collection of lessons can be created and named and printed out) so can I suggest to create a separate issue for this?

Also please comment in that ticket on whether law has any impact on us as a non-eu based project (QGIS is registered in Switzerland). We do plan to deploy under SSL regardless and having a mechanisme that course attendees can request their data to be removed sounds fine.
